src/cscout.cpp

```diff
@@ -2372,18 +2395,62 @@ vector<string> split_by_delimiter(string &s, char delim) {
 }
 
 
-
 // Produce call graphs with -R option
 static void produce_call_graphs(const vector <string> &call_graphs)
 {
 	char base_splitter = '?';
+	char opts_splitter = '&';
+	char opt_spltter = '=';
+	GDArgsKeys gdargskeys;
 
 	for (string url: call_graphs) {
 		vector<string> split_base_and_opts = split_by_delimiter(url, base_splitter);
-		for (string z: split_base_and_opts) {
-			cerr << z << endl;
+		if (split_base_and_opts.size() == 0) {
+			cerr << url << "is not a valid url" << endl;
+			continue;
+		}
+		FILE *target = fopen(split_base_and_opts[0].c_str() , "w+");
+		string base = split_base_and_opts[0];
+		GDTxt gd(target);
+		// Disable swill
+		gd.uses_swill = false;
+		vector<string> opts;
+
+		if (split_base_and_opts.size() == 1) goto call_graph_functions;
```

Comment:
Put the code below in a separate function  to avoid the goto.

Reply:
It is now in an if-statement